calico_node/startup/startup.go

```diff
@@ -190,26 +190,33 @@ func configureLogging() {
 // determineNodeName is called to determine the node name to use for this instance
 // of calico/node.
 func determineNodeName() string {
-	// Determine the name of this node.
-	nodeName := os.Getenv("NODENAME")
-	if nodeName == "" {
-		// NODENAME not specified, check HOSTNAME (we maintain this for
-		// backwards compatibility).
-		log.Info("NODENAME environment not specified - check HOSTNAME")
-		nodeName = os.Getenv("HOSTNAME")
-	}
-	if nodeName == "" {
-		// The node name has not been specified.  We need to use the OS
-		// hostname - but should warn the user that this is not a
-		// recommended way to start the node container.
-		var err error
-		if nodeName, err = names.Hostname(); err != nil {
-			log.WithError(err).Error("Unable to determine hostname")
-			terminate()
-		}
-		log.Warn("Auto-detecting node name. It is recommended that an explicit value is supplied using the NODENAME environment variable.")
+	var nodeName string
+	var err error
+
+	// Determine the name of this node.  Precedence is NODENAME, HOSTNAME, os.Hostname.
+	// We use the names.Hostname which lowercases and trims the name.
+	if nodeName = strings.TrimSpace(os.Getenv("NODENAME")); nodeName != "" {
+		log.Infof("Using NODENAME environment for node name")
+	} else if nodeName = strings.TrimSpace(os.Getenv("HOSTNAME")); nodeName != "" {
```

Comment:
Aw man, I thought we deprecated and removed HOSTNAME is v3.0

Reply:
Indeed!  If we had removed HOSTNAME then I have no hestitation in saying "NODENAME" should be correct.

However, the real issue is the HOSTNAME parm because that may be set by the OS.

Perhaps we should just adjust the HOSTNAME and os.Hostname, but leave NODENAME - under the assumption that if it's set it'll be correct.